src/lib/translators/mongo.ts

```diff
@@ -1683,21 +1719,40 @@ function transformTotals(step: Readonly<S.AddTotalRowsStep>): MongoStep {
     const aggs: { [id: string]: {} } = {};
     // get columns not in aggregation, i.e. columns that will hold the total rows labels
     const totalColumns: string[] = toCombine.filter(x => !comb.includes(x));
+    const countDistinctAddFields: PropMap<any> = {};
+    const addFieldsToAddToPipeline = [];
+
     for (const aggfStep of step.aggregations) {
       for (let j = 0; j < aggfStep.columns.length; j++) {
         const valueCol = aggfStep.columns[j];
         const aggregatedCol = aggfStep.newcolumns[j];
         const aggFunc = aggfStep.aggfunction;
-        aggs[aggregatedCol] = aggFunc == 'count' ? { $sum: 1 } : { [$$(aggFunc)]: $$(valueCol) };
+        if (aggFunc == 'count') {
+          aggs[aggregatedCol] = { $sum: 1 };
+        } else if (aggFunc == 'count distinct') {
+          // build a set of unique values
+          aggs[aggregatedCol] = { $addToSet: $$(valueCol) };
+          // count the number of items in the set
+          countDistinctAddFields[aggregatedCol] = { $size: $$(aggregatedCol) };
+        } else {
+          aggs[aggregatedCol] = { [$$(aggFunc)]: $$(valueCol) };
+        }
       }
     }
+
+    // if an addFields stage is not empty, make it ready to be inserted in the pipeline
+    if (Object.keys(countDistinctAddFields).length > 0) {
+      addFieldsToAddToPipeline.push({ $addFields: countDistinctAddFields });
+    }
```

Comment:
Suggestion : descendre la déclaration de `addFieldsToAddToPipeline` en dessous de la boucle, car non utilisée dedans et "oneliner" son affectation tant qu'à faire.

Idem pour les deux autres méthodes si dessus.

```suggestion
    const addFieldsToAddToPipeline = Object.keys(countDistinctAddFields).length
      ? [ { $addFields: countDistinctAddFields } ]
      : [];
```

Reply:
ah yes carrément mieux merci !

Reply:
> Aussi, comment ça se passe pour valider les query mongo générées ? Tu a testé ça à la main sur une base ?

Yes exactement. On pourrait viser mieux... Mais pour l'instant c'est comme ça :)